src/main/java/io/github/elytra/concrete/DefaultMarshallers.java

```diff
@@ -75,111 +76,118 @@
 	 * Aliases: i32, int32, int, integer
 	 */
 	public static final Marshaller<? extends Number> INT32 = weld(ByteBuf::writeInt, ByteBuf::readInt);
-	
-	
+
+
 	/**
 	 * Signed 64-bit (8 byte) integer.
 	 * <p>
 	 * Aliases: i64, int64, long
 	 */
 	public static final Marshaller<? extends Number> INT64 = weld(ByteBuf::writeLong, ByteBuf::readLong);
-	
-	
+
+
 	/**
 	 * 32-bit floating point.
 	 * <p>
 	 * Aliases: f32, float
 	 */
 	public static final Marshaller<? extends Number> FLOAT = weld(ByteBuf::writeFloat, ByteBuf::readFloat);
-	
-	
+
+
 	/**
 	 * 64-bit floating point.
 	 * <p>
 	 * Aliases: f64, double
 	 */
 	public static final Marshaller<? extends Number> DOUBLE = weld(ByteBuf::writeDouble, ByteBuf::readDouble);
-	
-	
+
+
 	/**
 	 * Protobuf variable sized integer.
 	 * <p>
 	 * Aliases: varint
 	 */
 	public static final Marshaller<? extends Number> VARINT = new VarIntMarshaller();
-	
-	
+
+
 	/**
 	 * Compound NBT tag.
 	 */
 	public static final Marshaller<NBTTagCompound> NBT = weld(ByteBufUtils::writeTag, ByteBufUtils::readTag);
-	
-	
+
+
 	/**
 	 * 64-bit packed BlockPos.
 	 */
 	public static final Marshaller<BlockPos> BLOCKPOS = new BlockPosMarshaller();
-	
-	
+
+
 	/**
 	 * UTF-8 varint-length-prefixed string.
 	 */
 	public static final Marshaller<String> STRING = weld(ByteBufUtils::writeUTF8String, ByteBufUtils::readUTF8String);
-	
+
+
 	/**
 	 * Packed ItemStack.
 	 */
 	public static final Marshaller<ItemStack> ITEMSTACK = weld(ByteBufUtils::writeItemStack, ByteBufUtils::readItemStack);
-	
-	
-	
+
+
+	/**
+	 * A bytebuf, only writes the readable bytes.
+	 */
+	public static final Marshaller<? extends ByteBuf> BYTEBUF = new ByteBufMarshaller();
+
+
 	private static final Map<String, Marshaller<?>> byName = Maps.newHashMap();
-	
-	
+
+
 	static {
 		put(UINT8, "u8", "uint8", "ubyte");
 		put(INT8, "i8", "int8", "byte");
-		
+
 		put(UINT16, "u16", "uint16", "char", "ushort");
 		put(INT16, "i16", "int16", "short");
-		
+
 		put(UINT24, "u24", "uint24", "umedium");
 		put(INT24, "i24", "int24", "medium");
-		
+
 		put(UINT32, "u32", "uint32", "uint", "uinteger");
 		put(INT32, "i32", "int32", "int", "integer");
-		
+
 		put(INT64, "i64", "int64", "long");
-		
+
 		put(FLOAT, "f32", "float");
 		put(DOUBLE, "f64", "double");
-		
+
 		put(VARINT, "varint");
-		
+
 		put(NBT, "nbt");
-		
+
 		put(BLOCKPOS, "blockpos");
-		
+
 		put(STRING, "string", "str", "utf8");
-		
+
 		put(ITEMSTACK, "item", "stack", "itemstack");
+
+		put(BYTEBUF, "bytebuf", "buf");
```

Comment:
This will require you to specify "bytebuf" in \@MarshalledAs - you should put it in the checks in getByType instead so it's a default.